test/index.js

```diff
@@ -1708,6 +1764,62 @@ describe('list', function() {
 
   });
 
+  describe('dropLast', function() {
+
+    it('is a binary function', function() {
+      eq(typeof S.dropLast, 'function');
+      eq(S.dropLast.length, 2);
+    });
+
+    it('type checks its arguments', function() {
+      assert.throws(function() { S.dropLast([1, 2, 3]); },
+                    errorEq(TypeError,
+                            '‘dropLast’ requires a value of type Number ' +
+                            'as its first argument; received [1, 2, 3]'));
+
+      assert.throws(function() { S.dropLast(0, null); },
+                    errorEq(TypeError,
+                            'The second argument to ‘dropLast’ ' +
+                            'cannot be null or undefined'));
+    });
+
+    it('returns a Nothing if n is greater than collection length', function() {
+      eq(S.dropLast(6, ['a', 'b', 'c', 'd', 'e']), S.Nothing());
+      eq(S.dropLast(6, 'abcde'), S.Nothing());
+    });
+
+    it('returns a Nothing if n is negative', function() {
+      eq(S.dropLast(-3, ['a', 'b', 'c', 'd', 'e']), S.Nothing());
+      eq(S.dropLast(-0, ['a', 'b', 'c', 'd', 'e']), S.Nothing());
+      eq(S.dropLast(-3, 'abcde'), S.Nothing());
+      eq(S.dropLast(-0, 'abcde'), S.Nothing());
+      // jshint -W053
+      eq(S.dropLast(new Number(-0), ['a', 'b', 'c', 'd', 'e']), S.Nothing());
+      // jshint +W053
+    });
+
+    it('returns an empty collection if n is equal to collection length', function() {
+      eq(S.dropLast(5, ['a', 'b', 'c', 'd', 'e']), S.Just([]));
+      eq(S.dropLast(5, 'abcde'), S.Just(''));
+    });
+
+    it('returns Just the first three elements from the collection', function() {
```

Comment:
We're now testing 6, 5, 2, and 0. For a collection of length five I'd like to test 0, 1, 2, 3, 4, 5, and 6. This may be a bit tedious, so we could choose a smaller collection length. If we choose three, for example, we could then test 0, 1, 2, 3, and 4.